persist/fs/seek.go

```diff
@@ -181,6 +181,13 @@ func (s *seeker) Open(namespace ts.ID, shard uint32, blockStart time.Time) error
 		return err
 	}
 
+	if !s.keepUnreadBuf {
```

Comment:
hmm? it was called in readIndex and readIndex is only called in Open(), I moved it here because if we call readIndex before readInfo, the readInfo call would panic on nil decoder as well, so it's safer to call it here?